include/envoy/secret/secret_provider.h

```diff
@@ -18,7 +19,17 @@ template <class SecretType> class SecretProvider {
    */
   virtual const SecretType* secret() const PURE;
 
-  // TODO(lizan): Add more methods for dynamic secret provider.
+  /**
+   * Add secret callback into secret provider.
```

Comment:
Can you add some comments on thread safety? E.g. from which threads is it safe to call this, on which thread will the callback be invoked?

Reply:
Comments are added. Thanks.